### PR TITLE
[Request access] Add filter by status UI

### PR DESCRIPTION
**Pull request by erzhtor**
> **Warning** Blocked by https://github.com/sourcegraph/sourcegraph/pull/50012 as it was made on top of this PR branch to avoid merge conflicts.

Closes https://github.com/sourcegraph/sourcegraph/issues/48255.

## Test plan
- `sg start`
- Disable built-in signup to make request access enabled
- Submit some requests if there are no requests yet https://sourcegraph.test:3443/request-access
- Open https://sourcegraph.test:3443/site-admin/account-requests

## Screenshots
| Before | After |
| --: | :-- |
| image | image | 
| n/a | image | 
| n/a | image | 

## App preview:

- [Web](https://sg-web-erzhtor-add-account-request.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/site-admin/AccessRequestsPage/index.tsx [hunk 1]
```diff
@@ -225,21 +232,48 @@ export const AccessRequestsPage: React.FunctionComponent = () => {
...
+                                {
+                                    key: 'Approve',
+                                    label: 'Approve',
+                                    icon: mdiCheck,
+                                    onClick: ([item]) => handleApprove(item.id, item.name, item.email),
+                                    // Can approve if there are remaining seats and not already approved
+                                    // We allow to approve previously rejected requests too
+                                    condition: () => hasRemainingSeats && status !== AccessRequestStatus.APPROVED,
+                                },
+                            ]}
```
Comment (kopancek) [inline client/web/src/site-admin/AccessRequestsPage/index.tsx:270]: What was the reason to change this to hidden menu as opposed to previous behavior where admin could just click once? This way the actions are quite hidden...

**Review by kopancek — APPROVED** [review]
The table view looks more condensed and harder to read, at least for me. Was that intentional or should we add some padding to each row?

File: client/web/src/site-admin/AccessRequestsPage/index.tsx [hunk 2]
```diff
@@ -225,21 +232,48 @@ export const AccessRequestsPage: React.FunctionComponent = () => {
...
-                        totalCount={connection?.totalCount ?? null}
-                        totalLabel={connection?.totalCount === 1 ? 'account request' : 'account requests'}
-                        {...paginationArgs}
-                    />
+                <div className="d-flex align-items-start justify-content-between">
+                    <AccessRequestStatusPicker status={status as AccessRequestStatus} onChange={setStatus} />
+                    <div className="d-flex justify-content-end mt-4">
+                        <PageSwitcher
+                            totalCount={connection?.totalCount ?? null}
+                            totalLabel={connection?.totalCount === 1 ? 'account request' : 'account requests'}
```
Comment (sashaostrikov) [inline client/web/src/site-admin/AccessRequestsPage/index.tsx:240]: [`pluralize`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/common/src/util/strings.ts?L24-26) can be used here
```suggestion
                            totalLabel={pluralize('account request', connection?.totalCount || 0)}
```

File: client/web/src/hooks/useURLSyncedString.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { useSearchParams } from 'react-router-dom'
+
+/**
+ * "useState" like hook that syncs primitive string state with URL search parameters.
+ *
+ * @param defaultValue initial data to use if no URL search parameter is set
+ */
+export function useURLSyncedString(key: string, defaultValue: string): [string, (newValue: string) => void] {
```
[the remark this answers is not in the document]
Comment (erzhtor) [reply client/web/src/hooks/useURLSyncedString.ts:8]: I intentionally created this version, as it is simpler for one param case. Even when multiple params, it is still nicer to have multiple independent setters than one reducer like a object setter.
```tsx
const [params, setParams] = useURLSyncedState({'foo': 'foo-value', 'bar': 'bar-value'})
const handleKey1Change = (value) => setParams({'foo', value})
const handleKey2Change = (value) => setParams({'bar', value})
// vs
const [key1, setKey1] = useURLSyncedString('foo', 'foo-value')
const [key2, setKey2] = useURLSyncedString('bar', 'bar-value')
```

File: client/web/src/site-admin/AccessRequestsPage/index.tsx (hunk 1 above)
Comment (erzhtor) [reply client/web/src/site-admin/AccessRequestsPage/index.tsx:270]: I thought to make pages consistent by reusing the Table + actions from [UserManagement page](https://sourcegraph.sourcegraph.com/site-admin/users?searchText=&offset=0&limit=25&orderBy=EVENTS_COUNT&descending=false&isAdmin=&maxEventsCount=&lastActiveAt=&createdAt=&deletedAt=%7B%22isNegated%22%3Atrue%7D). But now thinking back, I agree that having immediately visible buttons would be better than having hidden inside dropdown actions.

File: client/web/src/hooks/useURLSyncedString.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (sashaostrikov) [reply client/web/src/hooks/useURLSyncedString.ts:8]: yep, makes sense! 👍🏻

**Review by sashaostrikov — APPROVED** [review]
LGTM!

**Review by kopancek — APPROVED** [review]
LGTM as well, thanks for making the changes 👍

**Pull request merged**